Bug 1 - Smithy Card
My card function tests managed to find a few bugs right off the bat. In the smithy card, everything worked as expected, with the exception that the discard pile remained unchanged. My tests verified that the player was receiving the correct number of cards in their hand, and all other functionality was correct, so I assumed that the card must have been trashed instead of discarded. A closer look into the playSmithy() function revealed that this was indeed the case, and that in the discardCard() function the last parameter passed was 0-1 instead of 0. This verified my suspicions as the first line in discardCard trashes the card if that parameter is less than 0.

Bug 2 - Adventurer Card
Once again, A bug immediately revealed itself from my unit tests. After playing the adventurer card, the player should gain two treasure cards, then discard their adventurer card. A closer look at my test results shows that the player gained three cards in their hand at the end of their turn instead of two. The original bug where the game crashes if there are no treasure cards was also still there, but I assumed that didn’t count in this case as it was there before the assignment. After checking my group mate’s playAdventurer() function, I verified my suspicion of where the bug was contained. In the beginning of the function, the while loop loops until three treasure cards are obtained as opposed to only two.